src/runtime/update-component.ts

```diff
@@ -32,12 +32,35 @@ export const scheduleUpdate = (hostRef: d.HostRef, isInitialLoad: boolean) => {
   return BUILD.taskQueue ? writeTask(dispatch) : dispatch();
 };
 
-const dispatchHooks = (hostRef: d.HostRef, isInitialLoad: boolean) => {
+/**
+ * Dispatch initial-render and update lifecycle hooks, enqueuing calls to
+ * component lifecycle methods like `componentWillLoad` as well as
+ * {@link updateComponent}, which will kick off the virtual DOM re-render.
+ *
+ * @param hostRef a reference to a host DOM node
+ * @param isInitialLoad whether we're on the initial load or not
+ * @returns an empty Promise which is used to enqueue a series of operations for
+ * the component
+ */
+const dispatchHooks = (hostRef: d.HostRef, isInitialLoad: boolean): Promise<void> => {
   const elm = hostRef.$hostElement$;
   const endSchedule = createTime('scheduleUpdate', hostRef.$cmpMeta$.$tagName$);
-  const instance = BUILD.lazyLoad ? hostRef.$lazyInstance$ : (elm as any);
+  const instance = BUILD.lazyLoad ? hostRef.$lazyInstance$ : elm;
+
+  // We're going to use this variable together with {@link enqueue} to implement a
```

Comment:
Non-blocking nit - these JSDoc tags (`{@link}`) don't necessarily get detected in editors when using single line comments